### ISSUE TITLE
Search block design inconsistency

### ISSUE THREAD

**Issue by ddewan-git** [WordPress/twentytwentyfive#384]
When a "Button only" search control is expanded, there is no gap between the search (or search icon) button and the input field. I'm not sure if this no-gap design choice is intentional, but the style looks odd. It's also inconsistent with [the search forms](https://www.figma.com/design/dzGCSntVch4EQdVERTqyVK/Twenty-Twenty-Five?node-id=3112-235) used elsewhere, e.g. Search Results template.

![Search Block Test – Search Label Off – Button Only - Front-end - Current](https://github.com/user-attachments/assets/0367b10a-26c7-49ea-809d-484346b41fe8)

**Steps to Reproduce**
- Add the Search block to any post/page.
- Use the toggle search label button to hide the label.
- Choose the "Button only" option.
- Update and visit the post/page.
- Click on the Search button.

**Expected Result**
When a user clicks on the search (or search icon) button to expand the Search input text field, there should be a gap between the items.

![Search Block Test – Search Label Off – Button Only - Front-end - Expected Result](https://github.com/user-attachments/assets/54ddc19b-ba6c-46b2-9b6c-658a91a848d6)

**Comment by carolinan** [comment]
Thank you for the report.  I tested it and found that this is a problem with the block itself and it is not limited to this theme. It happens in every theme that I tested, both classic and block themes, default themes and third-party themes.

There is CSS added to the block when this settings is enabled, and the spacing added to the block in theme.json is overwritten:

```
.wp-block-search.wp-block-search__button-only .wp-block-search__button {
    flex-shrink: 0;
    margin-left: 0;
    max-width: 100%;
}
```

Therefor, the problem needs to be fixed in the editor, not the theme. The first step is to open the issue in the Gutenberg GitHub repository.
